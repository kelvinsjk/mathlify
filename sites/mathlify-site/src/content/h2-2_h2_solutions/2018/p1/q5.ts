import { Answer } from '$lib/classes/answer';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { Polynomial, quotient } from 'mathlify';
import { EquationWorking, ExpressionWorking } from 'mathlify/working';
import { QED } from '$lib/typesetting/utils';

export const answer = new Answer();

// ! Complete, but can be further abstracted

// (x+a)(x+b)
// x

const num = new Polynomial([1, 'a']);
const den = new Polynomial([1, 'b']);
const f = quotient(num, den);
const g = Polynomial.ofDegree(1);
const verbatim = true;

{
	const working = new ExpressionWorking(f.subIn({ x: f }, { verbatim }), { leadingEqual: true });
	working.combineFraction().simplify();
	const working2 = new EquationWorking(working.expression, g);
	working2.crossMultiply().expand().toPolynomial('x', { target: 'b' });
	const b = -1;
	const fInv = f.subIn({ b });
	const soln = mathlifier`$${'align*'} ff(x) &= f\\left( ${f} \\right) \\\\ ${working}

Since ${'ff = g'},
$${'gather*'} ${working2}

Comparing coefficients of ${'x^2'},
$${'align*'} 1+b &= 0 \\\\ b &= ${b} \\; ${QED}

$${'align*'} 
ff(x) &= g(x)
\\\\ ff(x) &= x
\\\\ f(x) &= f^{-1}(x)
\\\\ f^{-1}(x) &= f(x)
\\\\ &= ${f}
\\\\ &= ${fInv} \\; ${QED}
`;
	const ans = mathlifier`${{}} b = ${b}.
\\
${{}} f^{-1}(x) = ${fInv}.`;
	answer.addBody(ans, soln);
}
