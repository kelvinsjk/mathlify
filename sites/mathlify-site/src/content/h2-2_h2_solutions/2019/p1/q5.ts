import { Answer } from '$lib/classes/answer';
import { Expression, expTerm, Polynomial, sum } from 'mathlify';
import { expInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { generateAns as solveFgx } from '$content/h2-1_h2_learn/01_functions/03_composite/05_compose-inverse/02_practice';

export const answer = new Answer();

// ! Complete

// e^(2x) - 4
// x + 2

const [a, b, c] = [2, -4, 2];
const ax = Polynomial.ofDegree(1, { coeff: a });
const f = sum(expTerm(ax), b);
//const g = new Polynomial([1, c]);
const yMinusB = sum('y', -b);

// i
let fInv: Expression;
{
	const { ans, soln, exp } = expInverse(f, ax, yMinusB, { b });
	fInv = exp;
	answer.addPart(ans, soln);
}
// ii
{
	const x = 5;
	const { ans, soln } = solveFgx({ a, b, c, rhs: x, fnType: 'exp' }, fInv, { qed: true });
	answer.addPart(ans, soln);
}
