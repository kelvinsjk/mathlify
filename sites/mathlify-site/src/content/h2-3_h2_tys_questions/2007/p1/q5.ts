/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Polynomial, quotient } from 'mathlify';
import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';

export const question = new Question();

// (2x+7)/(x+2)
const num = new Polynomial([2, 7]);
const den = new Polynomial([1, 2]);
const exp = quotient(num, den);

// a
{
	question.addPart(
		mathlifier`Show that the equation ${{}} {y = ${exp}}
can be rewritten as ${{}} {y= A + \\frac{B}{${den}},}
where ${'A'}
and ${'B'}
are constants to be found.

Hence state a sequence of transformations which transform the graph
of ${{}}{y=\\frac{1}{x}}
to the graph of ${{}}{y= ${exp}.}`,
		{ marks: 4 }
	);
}
// b
{
	question.addPart(
		mathlifier`Sketch the graph of ${{}} {y = ${exp},}
giving the equations of any asymptotes and the coordinates
of any points of intersection with the ${'x\\text{-}'}
and ${'y\\text{-axes}.'}`,
		{ marks: 3 }
	);
}
