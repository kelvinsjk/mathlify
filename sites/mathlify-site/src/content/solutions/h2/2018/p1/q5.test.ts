import { expect, it } from 'vitest';
import { answer } from './q5.soln';

it('2018p1q5', () => {
	expect(answer).toMatchSnapshot();
});
