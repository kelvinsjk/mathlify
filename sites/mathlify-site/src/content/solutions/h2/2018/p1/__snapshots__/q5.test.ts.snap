// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2018p1q5 1`] = `
Answer {
  "answer": {
    "body": "<p><math><mrow><mi>b</mi><mo>=</mo></mrow><mrow><mo form="prefix" stretchy="false">−</mo></mrow><mrow><mn>1</mn><mi>.</mi></mrow></math>
<br>
<math><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mfrac><mi>.</mi></mrow></math></p>
",
  },
  "solution": {
    "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>f</mi><mrow><mo fence="true" form="prefix">(</mo><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac><mo>+</mo><mi>a</mi></mrow><mrow><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac><mo>+</mo><mi>b</mi></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>a</mi><mi>x</mi><mo>+</mo><mi>a</mi><mi>b</mi></mrow><mrow><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>b</mi><mi>x</mi><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup></mrow><mrow><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>a</mi><mi>x</mi><mo>+</mo><mi>a</mi><mi>b</mi></mrow><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>b</mi><mi>x</mi><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup></mrow></mfrac></mrow></mtd></mtr></mtable></math></p>
<p>Since <math><mrow><mi>f</mi><mi>f</mi><mo>=</mo></mrow><mrow><mi>g</mi><mo separator="true">,</mo></mrow></math>
<math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>a</mi><mi>x</mi><mo>+</mo><mi>a</mi><mi>b</mi></mrow><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>b</mi><mi>x</mi><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup></mrow></mfrac><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>a</mi><mi>x</mi><mo>+</mo><mi>a</mi><mi>b</mi><mo>=</mo><mi>x</mi><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>b</mi><mi>x</mi><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>+</mo><mi>a</mi><mo>+</mo><mi>a</mi><mi>x</mi><mo>+</mo><mi>a</mi><mi>b</mi><mo>=</mo><msup><mi>x</mi><mn>2</mn></msup><mo>+</mo><mi>x</mi><mi>a</mi><mo>+</mo><msup><mi>x</mi><mn>2</mn></msup><mi>b</mi><mo>+</mo><mi>x</mi><msup><mi>b</mi><mn>2</mn></msup></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo>+</mo><mi>a</mi><mo fence="true" form="postfix">)</mo></mrow><mi>x</mi><mo>+</mo><mi>a</mi><mi>b</mi><mo>+</mo><mi>a</mi><mo>=</mo><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo>+</mo><mi>b</mi><mo fence="true" form="postfix">)</mo></mrow><msup><mi>x</mi><mn>2</mn></msup><mo>+</mo><mrow><mo fence="true" form="prefix">(</mo><mi>a</mi><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup><mo fence="true" form="postfix">)</mo></mrow><mi>x</mi></mrow></mtd></mtr></mtable></math></p>
<p>Comparing coefficients of <math><mrow><msup><mi>x</mi><mn>2</mn></msup><mo separator="true">,</mo></mrow></math>
<math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mn>1</mn><mo>+</mo><mi>b</mi></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mn>0</mn></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mi>b</mi></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mn>1</mn><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>x</mi></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msup><mi>f</mi><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mfrac><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></mtd></mtr></mtable></math></p>
",
  },
}
`;
